dbt_subprojects/daily_spellbook/models/_metrics/transactions/chains/ton/metrics_ton_transactions_daily.sql

```diff
@@ -0,0 +1,66 @@
+{{ config(
+        schema = 'metrics_ton'
+        , alias = 'transactions_daily'
+        , materialized = 'incremental'
+        , file_format = 'delta'
+        , incremental_strategy = 'merge'
+        , unique_key = ['blockchain', 'block_date']
+        , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_date')]
+        )
+}}
+
+with ton_prices as ( -- get price of TON for each day to estimate USD value
+    select
+        date_trunc('day', minute) as price_day
+        , avg(price) as price
+    from {{ source('prices', 'usd') }}
+    where true
+        and symbol = 'TON'
+        group by 1
+), jetton_prices as (
+    -- jetton prices based on onchain data, see https://github.com/ton-studio/dune-queries?tab=readme-ov-file#duneton_foundationresult_jetton_price_daily
+    select jp.token_address as jetton_master, ts as block_date, avg(price_usd) as price_usd
+    from dune.ton_foundation.result_jetton_price_daily jp
```

Comment:
```suggestion
    from {{ source("ton_foundation", "result_jetton_price_daily", database="dune") }}
```

Reply:
proper jinja formatting here.
then you'll need to establish in a source file, like this directory of sources (probably a new directory):
https://github.com/duneanalytics/spellbook/tree/main/sources/_datasets

follow up question:
how is this table built and maintained? we don't have a ton of usage of team owned matviews / data uploads in spellbook, as there is risk in the table disappearing and failing prod jobs here. it would be good to have an understanding behind it before approving

Reply:
@jeff-dude this table is based on the query saved in [this repo](https://github.com/ton-studio/dune-queries). I totally agree that it is not proper way to relay on something that can be changed easily. I can add this table as a separate model to the ``ton`` schema - what do you think about that approach?

Reply:
it would be lovely if you could recreate in spellbook, then use downstream in these models 🙏 
you can include in this PR for simplicity.
assumption here is that all data you need is on dune, so building this model here instead will be straightforward

Reply:
@jeff-dude added additional model with prices and switched from the ton_foundation materialised view. Please take a look